### PR TITLE
fix regex for arp query

### PR DESCRIPTION
**Pull request by nbulat**
Fix for get_arp_table, tested on CE5880

### CODE REVIEW

File: napalm_ce/ce.py [hunk 1]
```diff
@@ -768,7 +768,7 @@ def get_mac_address_table(self):
         mac_address_table = []
         command = 'display mac-address'
         output = self.device.send_command(command)
-        re_mac = r"(?P<mac>\S+)\s+(?P<vlan>\d+|-)\S+\s+(?P<interface>\S+)\s+(?P<type>\w+)\s+(?P<age>\d+|-)"
+        re_mac = r"(?P<mac>\S+)\s+(?P<vlan>\d+|-)\s+(?P<interface>\S+)\s+(?P<type>\w+)\s+(?P<age>\d+|-)"
```
Comment (thddaniel) [inline napalm_ce/ce.py]: Hi @nbulat This is get_mac_address_table instead of get_arp_table. Do you have any example of CE5880 output?

**Comment by nbulat** [comment]
Hi Hao,

Sorry about the confusion, yes, you're right - the fix is about
get_mac_address_table...
Following is the output of "display mac-address":
<CE5855>display mac-address
Flags: * - Backup
       # - forwarding logical interface, operations cannot be performed
based
           on the interface.
Age : dynamic MAC learned time in seconds
-------------------------------------------------------------------------------
MAC Address    VLAN          Learned-From        Type                Age
-------------------------------------------------------------------------------
0025-90d3-f2db 1             GE1/0/1             dynamic            4453
-------------------------------------------------------------------------------
Total items: 1

Following is the output on CE5855 without fix:

# salt -v 'ce5855' net.arp
Executing job with jid 20210719120037351349
-------------------------------------------

ce5855:
    ----------
    comment:
    out:
    result:
        True

Obviously, there is a multi-column output and there is a typo in regex,
expecting arbitrary no-whitespace string where whitespace should be present.
With the proposed fix:

# salt -v 'ce5855' net.mac
ce5855:
    ----------
    comment:
    out:
        |_
          ----------
          active:
              True
          interface:
              GE1/0/1
          last_move:
              -1.0
          mac:
              00:25:90:D3:F2:DB
          moves:
              -1
          static:
              False
          vlan:
              1
    result:
        True

Best regards,

Nenad Bulatovic

**Comment by thddaniel** [comment]
Hi @nbulat please update code to  `re_mac = r"(?P<mac>\S+)\s+(?P<vlan>\d+|-)\S*\s+(?P<interface>\S+)\s+(?P<type>\w+)\s+(?P<age>\d+|-)"`

Change `\S+` to  `\S*`, because the output also could be in the form of following:
[display_mac-address.txt](https://github.com/napalm-automation-community/napalm-ce/blob/master/test/unit/ce/mock_data/display_mac-address.txt)

VLAN/VSI/BD
100/-/-

**Comment by nbulat** [comment]
Added more universal regex that should hopefully work for various CEs.

**Comment by nbulat** [comment]
Hi Hao Tang,

Thank you for your guidance.
I have added \S* to mac regex...

Best regards,

Nenad Bulatović

**Comment by thddaniel** [comment]
Hi @nbulat Thanks for your contribution。

**Pull request merged**